include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
```

Comment:
In old versions of `libc++`, the `isnan` function is implemented using `std::enable_if<std::is_floating<T>::value, bool>`, but newer versions use `std::enable_if<std::is_arithmetic<T>::value, bool>`, which also supports integers.

I did some investigation here, but came up with no useful check for detecting a useful check for distinguishing between old and new versions of `libc++`.  I am confused about why they designed the library like this, but as far as I can tell, the `_LIBCPP_VERSION` macro always evaluates to `1101`.  (I tried checking out the `libc++` source from their svn repo and compared my version with the current trunk.)  There might be a "proper" solution here, but I don't know what it is.